src/core/src/op/broadcast.cpp

```diff
@@ -95,12 +95,18 @@ ov::PartialShape get_result_shape_bidirectional(const ov::Node* this_ptr,
         const auto& arg_shape_dim = arg_shape[i].get_length();
         const auto& target_shape_dim = target_shape[i].get_length();
         NODE_VALIDATION_CHECK(this_ptr,
-                              arg_shape_dim == 1 || target_shape[i] == 1 || arg_shape_dim == target_shape_dim,
+                              arg_shape_dim == 1 || target_shape_dim == 1 || arg_shape_dim == target_shape_dim,
                               "Broadcast incorrect target shape. Expecting either 1 or ",
                               arg_shape_dim,
                               ". Got ",
                               target_shape[i]);
-        result_shape[i] = std::max(arg_shape_dim, target_shape_dim);
+        if (arg_shape_dim == 1) {
```

Comment:
could you add a comment about numpy broadcast rules?
e.g.
 0 is not a special case in numpy broadcasting, if we broadcast 1 and 0 dims, the resulting dim is 0.